mobgap/initial_contacts/evaluation.py

```diff
@@ -72,6 +77,67 @@ def calculate_matched_icd_performance_metrics(
     return icd_metrics
 
 
+def calculate_matched_icd_error(
+    ic_list_detected: pd.DataFrame, ic_list_reference: pd.DataFrame, matches: pd.DataFrame, sampling_rate_hz: float
+) -> dict[str, Union[float, int]]:
+    """
+    Calculate error metrics for initial contact detection results.
+
+    This function assumes that you already classified the detected initial contacts as true positive (tp), false
+    positive (fp), or false negative (fn) matches using the
+    :func:`~mobgap.initial_contacts.evaluation.categorize_ic_list` function.
+    The dataframe returned by categorize function can then be used as input to this function.
+
+    The following metrics are calculated for each true positive initial contact:
+
+    - `ic_absolute_error_s`: Absolute time difference (in seconds) between the detected and reference initial contact.
+    - `ic_relative_error`: All absolute errors, within a walking bout, divided by the average step duration estimated
+     by the INDIP.
+
+    Parameters
+    ----------
+    ic_list_detected: pd.DataFrame
+        The dataframe of detected initial contacts.
+    ic_list_reference: pd.DataFrame
+        The ground truth initial contact dataframe.
+    matches: pd.DataFrame
+        A dataframe containing the matches between detected and reference initial contacts as output
+        by :func:`~mobgap.initial_contacts.evaluation.evaluate_initial_contact_list`.
+    sampling_rate_hz: float
+        Sampling rate of the data.
+
+    Returns
+    -------
+    error_metrics: dict
+
+    """
+    # matching ics (within the window)
+    match_ics = get_matching_ics(
+        metrics_detected=ic_list_detected,
+        metrics_reference=ic_list_reference,
+        matches=matches,
+    )
+
+    # calculate absolute error in seconds
+    ic_absolute_error_s = abs(match_ics["ic"]["detected"] - match_ics["ic"]["reference"]) / sampling_rate_hz
+
+    # relative error (estimated by dividing all absolute errors, within a walking bout, by the average step duration
+    # estimated by the INDIP)
+    mean_ref_step_s = (
+        ic_list_detected.groupby(level="wb_id")["ic"].diff().dropna().groupby(level="wb_id").mean() / sampling_rate_hz
+    )
+
+    ic_relative_error = ic_absolute_error_s / mean_ref_step_s
```

Comment:
I think this is not the definition we use for relative error. We differentiate between abs rel error and rel error and in both cases I think you would usually calculate the relative value before taking the mean . `pipeline._error_metrics`. Might be useful to also use the functions from there?

Reply:
Thought about this further and mz command was stupid. The way you calculate it is fine. It is not a "normal" relative error. But we definitely need a better name for it to avoid confusion.